### PR TITLE
DP-5317 use GitHub reusable workflows to simplify yaml files

### PR DESCRIPTION
**Pull request by TimoSairiala**
Uses combination of github reusable workflow and product matrix to simplify px4 building. Here we can define products in fc_matrix and it will call tiiuae-pixhawk-and-saluki-builder.yaml with each product

### CODE REVIEW

File: .github/workflows/tiiuae-pixhawk-and-saluki.yaml [hunk 1]
```diff
@@ -9,276 +9,33 @@ on:
...
-        uses: actions/upload-artifact@v3
-        with:
-          name: pixhawk
-          path: bin/
-          retention-days: 1
-
-  saluki-v1_bootloader:
-    name: build saluki-v1_bootloader
-    runs-on: ubuntu-latest
-    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push' || github.event_name == 'workflow_dispatch'
```
Comment (TimoSairiala) [inline .github/workflows/tiiuae-pixhawk-and-saluki.yaml:66]: This if I don't understand. It is a bit complex but it seems to check if this is eighter pull_request, push or workflow_diplatch and only run this step if one of those is true. But this workflow is only triggered for those events. So I didn't include this condition, but maybe there is something I don't see now?

File: .github/workflows/tiiuae-pixhawk-and-saluki.yaml [hunk 2]
```diff
@@ -9,276 +9,33 @@ on:
...
-        with:
-          name: pixhawk
-          path: bin/
-          retention-days: 1
-
-  # NOTE: private repo is needed for this build, which is not available for forked PRs
-  saluki-v1_default:
-    name: build saluki-v1_default
-    runs-on: ubuntu-latest
-    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push' || github.event_name == 'workflow_dispatch'
```
Comment (TimoSairiala) [inline .github/workflows/tiiuae-pixhawk-and-saluki.yaml:40]: This if I don't understand. It is a bit complex but it seems to check if this is eighter pull_request, push or workflow_diplatch and only run this step if one of those is true. But this workflow is only triggered for those events. So I didn't include this condition, but maybe there is something I don't see now?

File: .github/workflows/tiiuae-pixhawk-and-saluki.yaml [hunk 3]
```diff
@@ -9,276 +9,33 @@ on:
...
-        uses: actions/checkout@v3
-        with:
-          path: px4-firmware
-          fetch-depth: 0
-      - name: Run px4-firmware pixhawk build
-        run: |
-          set -eux
-          mkdir -p bin
-          cd px4-firmware/
-          ./clone_public.sh
```
Comment (TimoSairiala) [inline .github/workflows/tiiuae-pixhawk-and-saluki.yaml:26]: here is ran clone_public.sh but in other steps we don't run this. In other steps we are cloning submodules instead. Maybe there is a reason for this, so I have kept it this way with reusable workflow as well. Controlled by clone_public flag

Comment (jnippula) [reply .github/workflows/tiiuae-pixhawk-and-saluki.yaml:26]: clone_public skips all private submodules, so this action for pixhawk can be run also in forked repos.

**Review by jnippula — APPROVED** [review]

File: .github/workflows/tiiuae-pixhawk-and-saluki.yaml (hunk 2 above)
Comment (TimoSairiala) [reply .github/workflows/tiiuae-pixhawk-and-saluki.yaml:40]: added back

File: .github/workflows/tiiuae-pixhawk-and-saluki-builder.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - name: Checkout px4-firmware
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_REPO_TOKEN }}
+          # if we clone_public, then do not clone submodules, otherwise 'recursive'
+          submodules: ${{ inputs.clone_public == true && 'false' || 'recursive' }}
+          path: px4-firmware
+          fetch-depth: 0
+      - name: Checkout PF Athena crypto driver
```
[the remark this answers is not in the document]
Comment (TimoSairiala) [reply .github/workflows/tiiuae-pixhawk-and-saluki-builder.yaml]: removed this

**Comment by TimoSairiala** [comment]
This branch had so many commits, so I squash merged it to main

**Pull request closed without merging**